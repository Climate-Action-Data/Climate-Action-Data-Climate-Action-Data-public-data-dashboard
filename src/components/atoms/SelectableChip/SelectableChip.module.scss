.selected {
  color: white;
  background-color: #00242C;
  cursor: pointer;
  border-radius: 32px;
  padding: 6px 12px;
}

.selected:hover {
  background-color: #3E4A4E;
}

.notSelected {
  color: #4D5C62;
  background-color: transparent;
  cursor: pointer;
  border-radius: 32px;
  padding: 6px 12px;
}

.notSelected:hover {
  background-color: #DBDEE0;
}