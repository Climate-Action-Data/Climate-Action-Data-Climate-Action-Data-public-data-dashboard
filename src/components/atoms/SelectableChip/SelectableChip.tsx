import { FC } from 'react'
import { Tag, TagLabel } from '@chakra-ui/react'
import styles from './SelectableChip.module.scss'
import { Aeonik } from '@/styles/fonts'

interface SelectableChipProps {
  label: string
  isSelected?: boolean
  onClick?: () => void
}

const SelectableChip: FC<SelectableChipProps> = ({ label, isSelected = false, onClick = () => null }) => {
  return (
    <Tag className={`${Aeonik.className} ${isSelected ? styles.selected : styles.notSelected}`} onClick={onClick}>
      <TagLabel>{label}</TagLabel>
    </Tag>
  )
}

export default SelectableChip
