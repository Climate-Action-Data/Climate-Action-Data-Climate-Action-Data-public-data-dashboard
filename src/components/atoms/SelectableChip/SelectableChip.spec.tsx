import { render, screen } from '@testing-library/react'
import SelectableChip from '@/components/atoms/SelectableChip/SelectableChip'
import userEvent from '@testing-library/user-event'

describe(`Selectable Chip`, () => {
  it(`renders correctly for unselected`, () => {
    const { container } = render(<SelectableChip label={`label`} />)
    expect(container).toMatchSnapshot()
  })

  it(`renders correctly for selected`, () => {
    const { container } = render(<SelectableChip label={`label`} isSelected />)
    expect(container).toMatchSnapshot()
  })

  it(`renders correctly for selected`, async () => {
    const tOnClick = jest.fn()
    render(<SelectableChip label={`label`} isSelected onClick={tOnClick} />)
    await userEvent.click(screen.getByText(`label`))
    expect(tOnClick).toHaveBeenCalled()
  })
})
