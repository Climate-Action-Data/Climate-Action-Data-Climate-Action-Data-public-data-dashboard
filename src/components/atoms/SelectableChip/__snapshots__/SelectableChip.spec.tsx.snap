// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Selectable Chip renders correctly for selected 1`] = `
<div>
  <span
    class="css-1qm7lvz"
    data-testid="test-id-label"
  >
    <span
      class="css-zvlevn"
    >
      label
    </span>
  </span>
</div>
`;

exports[`Selectable Chip renders correctly for unselected 1`] = `
<div>
  <span
    class="css-1qm7lvz"
    data-testid="test-id-label"
  >
    <span
      class="css-zvlevn"
    >
      label
    </span>
  </span>
</div>
`;
